ons_alpha/private_media/models.py

```diff
@@ -0,0 +1,217 @@
+            return False
+        # Use 'last_private_collection' - but only if it is still private
+        if self.last_private_collection and self.last_private_collection.path.startswith(
+            get_private_root_collection_path()
+        ):
+            self.collection = self.last_private_collection
+        else:
+            # Fallback to the root private collection
+            self.collection_id = get_private_root_collection_id()
+        return True
+
+    def move_to_public_collection(self) -> bool:
+        """
+        Set this object's collection to one that is 'Public', without saving any
+        changes. Returns a bool indicating whether there was any change (e.g. if
+        the object already belongs to a public collection, `False` is returned).
+        """
+        if self.collection.path.startswith(get_public_root_collection_path()):
+            return False
+        # Use 'last_public_collection' - but only if it is still public
+        if self.last_public_collection and self.last_public_collection.path.startswith(
+            get_public_root_collection_path()
+        ):
+            self.collection = self.last_public_collection
+        else:
+            # Fallback to the root public collection
+            self.collection_id = get_public_root_collection_id()
+        return True
+
+    def set_file_acls(self, acl_name: str) -> bool:
+        """
+        Attempts to set the ACL for all privacy-controlled files related to this
+        object (which could be one or many).
+
+        If all of the change requests are successful, the object's `acls_last_set`
+        datetime will be updated on completion, and `True` will  be returned.
+
+        If any of the change requests fail, the `acls_last_set` datetime will not
+        be updated, and `False` will be returned.
+        """
+        files = list(self.get_privacy_controlled_files())
+        failures = 0
+        for file in files:
+            if hasattr(file, "obj") and hasattr(file.obj, "Acl"):
+                try:
+                    file_acl = file.obj.Acl()
+                except ClientError:
+                    # TODO: differentiate between general connection failures
+                    # and other issues that couldn't possibly result in success
+                    logger.exception("Failed to retreive ACL for file: %s.", repr(file))
+                    failures += 1
+                    continue
+                try:
+                    file_acl.put(ACL=acl_name)
+                except ClientError:
+                    # TODO: differentiate between general connection failures
+                    # and other issues that couldn't possibly result in success
+                    logger.exception("Failed to set ACL for file: %s.", repr(file))
+                    failures += 1
+                    continue
+
+        if failures:
+            return False
+
+        self.acls_last_set = timezone.now()
```

Comment:
Nit: if `files` is empty, this is still updated.

Reply:
I think that's okay. The purpose of `acls_last_set` is to work out whether setting of ACLs still needs to take place. If no files need to be updated at all, there are also no remaining acls to set, so that's fine.